provider.go

```diff
@@ -42,6 +42,8 @@ type WasmcloudProvider struct {
 	putTargetLinkFunc func(InterfaceLinkDefinition) error
 	delSourceLinkFunc func(InterfaceLinkDefinition) error
 	delTargetLinkFunc func(InterfaceLinkDefinition) error
+	sourceLinkArr     []InterfaceLinkDefinition
+	targetLinkArr     []InterfaceLinkDefinition
```

Comment:
Instead of having a separate map, I think we can keep storing these just as source/targetLinks and initialize them later